### PR TITLE
Explain subprocess signal exit codes

### PR DESCRIPTION
**Pull request by jaytmiller**
Added explanation of negative subprocess return codes which correspond to UNIX
  signals which kill processes.  Reporting the signal status associated with a
  killed process is roughly equivalent to getting the status reported in the
  aftermath of a core dump.
Re-order Dockerfile and use developer user for pip installs to avoid pip complaints
  and improve typical build times.
Add "ulimit -a" dump to caldp-process to get the container's perception of various
  process limits,  including core dump configuration;  this may be fictional and/or
  require additional configuration in Terraform to set the docker run --ulimit
  parameter.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -79,6 +74,13 @@ RUN ./caldp-install-fitscut   /usr/local && \
    echo "/usr/local/lib" >> /etc/ld.so.conf && \
    ldconfig
 
+USER developer
+RUN pip install --upgrade pip
+RUN pip install awscli boto3
+# RUN pip install jupyterlab
```
Comment (bhayden53) [inline Dockerfile:80]: can you remove the jupyterlab line completely?

File: Dockerfile [hunk 2]
```diff
@@ -79,6 +74,13 @@ RUN ./caldp-install-fitscut   /usr/local && \
    echo "/usr/local/lib" >> /etc/ld.so.conf && \
    ldconfig
 
+USER developer
```
Comment (bhayden53) [inline Dockerfile:77]: I'm a little worried about this as it used to be under root. What's the need for the change?

File: Dockerfile (hunk 1 above)
Comment (jaytmiller) [reply Dockerfile:80]: Yeah.   Possibly I can get rid of all of that just by making them caldp dependencies if they're not already.   I'll fool with it.

Comment (bhayden53) [reply Dockerfile:80]: in particular I'm a tiny bit worried now about overwriting something from the base env

File: Dockerfile (hunk 2 above)
Comment (jaytmiller) [reply Dockerfile:77]: pip was complaining about installing as root during the build.  AFAICT conda was installed as developer,  so I think we should honor the complaint and follow suit.

Comment (bhayden53) [reply Dockerfile:77]: Something this has raised for me though is why are we even upgrading pip at all? shouldn't we be leaving it alone?

Comment (jaytmiller) [reply Dockerfile:77]: I propose deleting the whole block.  I don't see any reason to upgrade pip these days either.   Everything else is already a CALDP dependency and installed automatically.

I checked the base image and conda is installed as developer.   So I think our installs as root that pip is now complaining about were always (harmless) mistakes.   I propose doing all conda installs as developer.

I think we may be vulnerable to pip changing the baseline environment due to what we install.   caldp has dependencies and if they're not installed or are the wrong version I think the base will be changed.   The default pip upgrade policy is "only-if-needed" but I don't see anything more conservative like "never".   Maybe walking through with very explicit installs and --no-deps but we can only skip our deps if they don't matter in ops.

Comment (bhayden53) [reply Dockerfile:77]: I have had some similar concerns about the caldp install floating around at the back of my mind, and hoping there was never a reason for them to fly up to the front...

Comment (bhayden53) [reply Dockerfile:77]: I concur with deleting the whole block btw

Comment (jaytmiller) [reply Dockerfile:77]: Reading through the Dockerfile,  it's scrambled in a few ways:
1.  yum is in two places
2. crds is in two places
3. certs are in two places
I think all of those can/should be unified into cohesive blocks to make it easier to understand.   I'll offer to do it now but maybe it should be ticketed so the risk is folded in with some other testing?   Or ignored.

Comment (bhayden53) [reply Dockerfile]: hmm. feels like a separate ticket. there is certainly a risk there of blocking these unrelated changes from getting merged.

Comment (jaytmiller) [reply Dockerfile]: OK,  think I'm done here then.

**Comment by jaytmiller** [comment]
Agreed to merge.

**Pull request merged**